be/src/exec/pipeline/operator.h

```diff
@@ -43,8 +43,14 @@ class Operator {
     // The operator should finish processing.
     // The method should be idempotent, because it may be triggered
     // multiple times in the entire life cycle
+    // finish function is used to finish the following operator of the current operator that encounters its EOS
+    // and has no data to push into its following operator.
     virtual void finish(RuntimeState* state) = 0;
 
+    // finish_backward is used to finish the preceding operator of the current operator that finishes in advance
+    // and never need input data to consume.
+    virtual void finish_backward(RuntimeState* state) { finish(state); }
+
```

Comment:
Why not merge those two methods into one? Derived class can have different meanings.

Reply:
two functions have different meanings:
1. finish func is invoked by the current operator to its following operator that the EOS is comming, but the following operator  should still process the buffered chunks inside.
2. finish_backward is invoked by the current operator to preceding operators that they should be finished, the buffered data is needless.